index.js

```diff
@@ -11,19 +11,42 @@ var chroma = require('chroma-js');
 
 exports.main = function(){
 	var maxTabs = simplePrefs.prefs.maxTabs;
-	var max = (maxTabs > 1) ? maxTabs : 10; // Max at least 2 tabs please
+	var max = (maxTabs > 1) ? maxTabs : 2; // Max at least 2 tabs please
 	var maxPrivateBrowsing = simplePrefs.prefs.maxPrivateBrowsing;
+    var maxOpenInNextWindow = simplePrefs.prefs.maxOpenInNextWindow;
+    var maxExcludePinnedTabs = simplePrefs.prefs.maxExcludePinnedTabs;
 
 	var button = ActionButton({
 		id: 'max-tabs-button',
-	  label: title,
+	    label: title,
 		icon: './icon.svg',
 		disabled: true
 	});
 
+    var calcTabsLen = function(win){
+        let len = win.tabs.length;
+        if(maxExcludePinnedTabs){
+            let pinned = 0;
+            for(let i = 0; i < len; i++){
+                if(win.tabs[i].isPinned)
+                    pinned++;
+            }
+            len -= pinned;
```

Comment:
All right, let me write my last words on this topic here. The readability of both versions is just the same, the difference is only in a slightly higher number of lines. No magic happens in the loop, nothing unusual. And as people spend much much more time executing the function than reading its source, loop version must win. This loop is a hot code and thus it's not over-optimizing. Also I hope this addon is not a contest for the shortest code.